lib/puppet/functions/hiera_common.rb

```diff
@@ -0,0 +1,67 @@
+require 'hiera_puppet'
+
+# Provides the block that defines the hiera, hiera_array, and hiera_hash functions. The only difference
+# between those functions is the merge type (:priority, :array, or :hash) passed to the Hiera lookup. The
+# definition contained here derives the merge type from the name of the function.
+#
+class Puppet::Functions::HieraCommon
```

Comment:
We probably do not want to have this functionality in this namespace as it will fail mysteriously if someone tries to call the function "hiera_common" from puppet, as the loader will try to load this file. The error will be very different that if the file did not exist in this directory. I think it should move to some other suitable place (any subnamespace of lib/puppet/functions/ is also subject to the same issue - so move it out of the way; I think we have a lib/puppet/hiera namespace - move it there instead).